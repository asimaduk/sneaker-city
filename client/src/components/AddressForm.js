import React, { Component } from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

class AddressForm extends Component {
    constructor(){
        super();

        this.state = {
            firstname: '',
            lastname: '',
            address1: '',
            address2: '',
            city: '',
            state: '',
            zip: '',
            country: ''
        }
    }

    componentDidMount = () => {
        //using window.address as a global mode of storing address info
        if(window.address){
            this.setState({
                firstname: window.address.firstname || '',
                lastname: window.address.lastname || '',
                address1: window.address.address1 || '',
                address2: window.address.address2 || '',
                city: window.address.city || '',
                state: window.address.state || '',
                zip: window.address.zip || '',
                country: window.address.country || ''
            })
        }
    }

    render(){
        const { address1, address2, city, country, firstname, lastname, state, zip } = this.state;
        return (
            <div>
                <Typography variant="h6" gutterBottom>
                    Shipping address
                </Typography>

                <Grid container spacing={3}>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            id="firstName"
                            name="firstName"
                            label="First name"
                            fullWidth
                            autoComplete="fname"
                            value={firstname}
                            onChange={(event)=> {
                                this.setState({firstname: event.target.value})
                                if(window.address){
                                    window.address.firstname = event.target.value;
                                }
                                else {
                                    window.address = {
                                        firstname: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            id="lastName"
                            name="lastName"
                            label="Last name"
                            fullWidth
                            autoComplete="lname"
                            value={lastname}
                            onChange={(event)=> {
                                this.setState({lastname: event.target.value});

                                if(window.address){
                                    window.address.lastname = event.target.value;
                                }
                                else {
                                    window.address = {
                                        lastname: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12}>
                        <TextField
                            required
                            id="address1"
                            name="address1"
                            label="Address line 1"
                            fullWidth
                            autoComplete="billing address-line1"
                            value={address1}
                            onChange={(event)=> {
                                this.setState({address1: event.target.value});

                                if(window.address){
                                    window.address.address1 = event.target.value;
                                }
                                else {
                                    window.address = {
                                        address1: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12}>
                        <TextField
                            id="address2"
                            name="address2"
                            label="Address line 2"
                            fullWidth
                            autoComplete="billing address-line2"
                            value={address2}
                            onChange={(event)=> {
                                this.setState({address2: event.target.value});

                                if(window.address){
                                    window.address.address2 = event.target.value;
                                }
                                else {
                                    window.address = {
                                        address2: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            id="city"
                            name="city"
                            label="City"
                            fullWidth
                            autoComplete="billing address-level2"
                            value={city}
                            onChange={(event)=> {
                                this.setState({city: event.target.value});

                                if(window.address){
                                    window.address.city = event.target.value;
                                }
                                else {
                                    window.address = {
                                        city: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField 
                            id="state" 
                            name="state" 
                            label="State/Province/Region" 
                            fullWidth
                            value={state}
                            onChange={(event)=> {
                                this.setState({state: event.target.value});

                                if(window.address){
                                    window.address.state = event.target.value;
                                }
                                else {
                                    window.address = {
                                        state: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            id="zip"
                            name="zip"
                            label="Zip / Postal code"
                            fullWidth
                            autoComplete="billing postal-code"
                            value={zip}
                            onChange={(event)=> {
                                this.setState({zip: event.target.value});

                                if(window.address){
                                    window.address.zip = event.target.value;
                                }
                                else {
                                    window.address = {
                                        zip: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <TextField
                            required
                            id="country"
                            name="country"
                            label="Country"
                            fullWidth
                            autoComplete="billing country"
                            value={country}
                            onChange={(event)=> {
                                this.setState({country: event.target.value});
                                
                                if(window.address){
                                    window.address.country = event.target.value;
                                }
                                else {
                                    window.address = {
                                        country: event.target.value
                                    }
                                }
                            }}
                        />
                    </Grid>
                    <Grid item xs={12}>
                    <FormControlLabel
                        control={<Checkbox color="secondary" name="saveAddress" value="yes" />}
                        label="Use this address for payment details"
                    />
                    </Grid>
                </Grid>
            </div>
        )
    }
}

export default AddressForm;