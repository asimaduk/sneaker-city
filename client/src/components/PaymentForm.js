import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

class PaymentForm extends Component {
    constructor(){
        super();

        this.state = {
            cardname: '',
            cardnumber: '',
            expirydate: '',
            cvv: ''
        }
    }

    componentDidMount = () => {
        //using window.paymentCard as a global mode of storing payment card info
        if(window.paymentCard){
            this.setState({
                cardname: window.paymentCard.name || '',
                cardnumber: window.paymentCard.number || '',
                expirydate: window.paymentCard.expiryDate || '',
                cvv: window.paymentCard.cvv || '',
            })
        }
    }

    render(){
        const { cardname, cardnumber, cvv, expirydate } = this.state;

        return (
            <div>
                <Typography variant="h6" gutterBottom>
                    Payment method
                </Typography>
                <Grid container spacing={3}>
                <Grid item xs={12} md={6}>
                    <TextField 
                        required 
                        id="cardName" 
                        label="Name on card" 
                        fullWidth
                        value={cardname} 
                        onChange={(event)=> {
                            const value = event.target.value;
                            if(/^[a-zA-Z ]+$/.test(value) || !value){
                                this.setState({cardname: value});

                                if(window.paymentCard){
                                    window.paymentCard.name = value;
                                }
                                else {
                                    window.paymentCard = {
                                        name: value
                                    }
                                }
                            }
                        }}
                    />
                </Grid>
                <Grid item xs={12} md={6}>
                    <TextField 
                        required 
                        id="cardNumber" 
                        label="Card number" 
                        fullWidth
                        value={cardnumber}
                        onChange={(event)=> {
                            const value = event.target.value;
                            if(value.length < 17 && ((/^[0-9]+$/.test(value))  || !value)){
                                this.setState({cardnumber: value});

                                if(window.paymentCard){
                                    window.paymentCard.number = value;
                                }
                                else {
                                    window.paymentCard = {
                                        number: value
                                    }
                                }
                            }
                        }}
                    />
                </Grid>
                <Grid item xs={12} md={6}>
                    <TextField 
                        required 
                        id="expDate" label="Expiry date" 
                        fullWidth 
                        value={expirydate}
                        onChange={(event)=> {
                            const value = event.target.value;
                            if(value.length < 8 && ((/^[0-9/]+$/.test(value))  || !value)){
                                this.setState({expirydate: value});

                                if(window.paymentCard){
                                    window.paymentCard.expiryDate = value;
                                }
                                else {
                                    window.paymentCard = {
                                        expiryDate: value
                                    }
                                }
                            }
                        }}
                    />
                </Grid>
                <Grid item xs={12} md={6}>
                    <TextField
                        required
                        id="cvv"
                        label="CVV"
                        helperText="Last three digits on signature strip"
                        fullWidth
                        value={cvv}
                        onChange={(event)=> {
                            const value = event.target.value;
                            if(value.length < 4 && ((/^[0-9]+$/.test(value))  || !value)){
                                this.setState({cvv: value});

                                if(window.paymentCard){
                                    window.paymentCard.cvv = value;
                                }
                                else {
                                    window.paymentCard = {
                                        cvv: value
                                    }
                                }
                            }
                        }}
                    />
                </Grid>
                <Grid item xs={12}>
                    <FormControlLabel
                        control={<Checkbox color="secondary" name="saveCard" value="yes" />}
                        label="Remember credit card details for next time"
                    />
                </Grid>
            </Grid>
            </div>
        )
    }
}

export default PaymentForm;