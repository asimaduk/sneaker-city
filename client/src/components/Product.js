import React from 'react';
import { Link } from 'react-router-dom';

import '../App.css';

//format release date of product
const formatReleaseDate = (timestamp) => {
    const date = new Date(timestamp);
    
    let hours = date.getHours();
    let minutes = date.getMinutes();
    let ampm = hours >= 12 ? 'PM' : 'AM';

    hours = hours % 12;
    hours = hours ? hours : 12; // the hour '0' should be '12'
    minutes = minutes < 10 ? '0'+minutes : minutes;

    let strTime = `${hours}:${minutes} ${ampm}`;

    let months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']

    return `Available ${months[date.getMonth()]} ${date.getDate()} at ${strTime}`;
}

const Product = (props) => {
    const product = props.product;
    
    return (
        product ? 
            <Link to={{pathname: `/products/${product.name.toLowerCase()}/${product.id}`, product: product}} className="Product-Div">
                <img src={product.image} alt={product.name} className="Product-Image"/>
                <div className="Product-Price">${product.price}</div> 
                <div className="Product-Details">
                    <div className="Product-Wrapper">
                        <h3 className="Product-Name">{product.name}</h3>
                        <h6 className="Product-Desc">{product.model}</h6>
                        <p className="Product-ReleaseDate">{formatReleaseDate(product.releaseDate)}</p>
                    </div>
                </div>
            </Link>
            :
            null
    )
}

export default Product;