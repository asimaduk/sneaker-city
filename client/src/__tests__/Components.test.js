import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';

import AddressForm from '../components/AddressForm';
import PaymentForm from '../components/PaymentForm';
import Product from '../components/Product';
import ProductList from '../components/ProductList';
import ReviewOrder from '../components/ReviewOrder';

//tests components for successful rendering

describe('AddressForm', () => {
    test('renders without crashing', () => {
        const component = create(<AddressForm />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('PaymentForm', () => {
    test('renders without crashing', () => {
        const component = create(<PaymentForm />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Product', () => {
    test('renders without crashing', () => {
        const component = create(<Product />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ProductList', () => {
    const component = create(<ProductList />); 
    const tree = component.toJSON();

    test('renders without crashing', () => {
        expect(tree).toMatchSnapshot();
    });
});

describe('ReviewOrder', () => {
    test('renders without crashing', () => {
        const component = create(<ReviewOrder />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});