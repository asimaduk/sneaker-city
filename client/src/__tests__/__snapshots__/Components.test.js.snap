// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm renders without crashing 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Shipping address
  </h6>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="firstName"
        >
          First name
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="fname"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="firstName"
            name="firstName"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="lastName"
        >
          Last name
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="lname"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="lastName"
            name="lastName"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="address1"
        >
          Address line 1
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="billing address-line1"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="address1"
            name="address1"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="address2"
        >
          Address line 2
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="billing address-line2"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="address2"
            name="address2"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="city"
        >
          City
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="billing address-level2"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="city"
            name="city"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="state"
        >
          State/Province/Region
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="state"
            name="state"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="zip"
        >
          Zip / Postal code
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="billing postal-code"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="zip"
            name="zip"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="country"
        >
          Country
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="billing country"
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="country"
            name="country"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-213 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-216"
              data-indeterminate={false}
              name="saveAddress"
              onChange={[Function]}
              type="checkbox"
              value="yes"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Use this address for payment details
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`PaymentForm renders without crashing 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Payment method
  </h6>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="cardName"
        >
          Name on card
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="cardName"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="cardNumber"
        >
          Card number
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="cardNumber"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="expDate"
        >
          Expiry date
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="expDate"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
          htmlFor="cvv"
        >
          CVV
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="cvv-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="cvv"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root Mui-required"
          id="cvv-helper-text"
        >
          Last three digits on signature strip
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-213 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-216"
              data-indeterminate={false}
              name="saveCard"
              onChange={[Function]}
              type="checkbox"
              value="yes"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Remember credit card details for next time
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Product renders without crashing 1`] = `null`;

exports[`ProductList renders without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 300,
        "justifyContent": "center",
      }
    }
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`ReviewOrder renders without crashing 1`] = `
Array [
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Order summary
  </h6>,
  <ul
    className="MuiList-root"
  >
    <li
      className="MuiListItem-root makeStyles-listItem-256 MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          Total
        </span>
      </div>
      <h6
        className="MuiTypography-root makeStyles-total-257 MuiTypography-subtitle1"
      >
        $ 
        0
      </h6>
    </li>
  </ul>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <h6
        className="MuiTypography-root makeStyles-title-258 MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Shipping
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
         
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
         
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <h6
        className="MuiTypography-root makeStyles-title-258 MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Payment details
      </h6>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Card type
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Visa
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Card holder
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Card number
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Expiry date
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
